frame/contracts/src/storage/meter.rs

```diff
@@ -416,24 +426,19 @@ where
 		info: &mut ContractInfo<T>,
 	) -> Result<DepositOf<T>, DispatchError> {
 		debug_assert!(self.is_alive());
-
 		let ed = Pallet::<T>::min_balance();
-		let mut deposit =
-			Diff { bytes_added: info.encoded_size() as u32, items_added: 1, ..Default::default() }
-				.update_contract::<T>(None);
-
-		// Instantiate needs to transfer at least the minimum balance in order to pull the
-		// deposit account into existence.
-		// We also add another `ed` here which goes to the contract's own account into existence.
-		deposit = deposit.max(Deposit::Charge(ed)).saturating_add(&Deposit::Charge(ed));
-		if deposit.charge_or_zero() > self.limit {
+
+		let code_info = CodeInfoOf::<T>::get(info.code_hash).ok_or(Error::<T>::CodeNotFound)?;
```

Comment:
The `code_info` should already be loaded as part of the `WasmBlob` when `charge_instantiate` is called. Maybe it makes sense to make `code_info` available on the `Executable` trait and pass it into `charge_instantiate`.